### DISCUSSION TITLE
Using array of error messages per one validation rule

### DISCUSSION THREAD

**Discussion by ElenaMazyrina** [react-hook-form/react-hook-form#6206] [Q&A]
Use react-hook-form version 7.11.1.

I have a simple control which is expected to show multiple error messages per one validation rule when it is invalid. When I register this control I provide an "validate" option with a custom validation function. From validate option contract I see that my custom validation function should return ValidateResult type. This type can be: Message | Message[] | boolean | undefined (where Message is string). So I assume that I can return array of error messages if control is invalid. But if I return not a string but array, I can't access my error messages anywhere: formState.errors.filedName.mesage is empty.

Is there any way to use array of error messages to show them all in the interface?

My simple component is below:

```
export const ControlWithErrorsExample = () => {
    const {
        register,
        formState: { errors },
    } = useForm({
        mode: 'onChange',
    });

    const { testValue: testValueError } = errors;

    const validateField = (v: string): ValidateResult => {
        if (v.length > 3) {
            return ['Incorrect value', 'Max length should be equal or less than 3'];
        }

        return undefined;
    };

    useEffect(() => {
        console.log(`I need to get access to my error messages array but got: ${testValueError?.message}`);
    }, [testValueError]);

    return <input type="text" {...register('testValue', { validate: (v) => validateField(v) })} />;
};
```
Thanks in advance

**Comment by bluebill1049** [comment]
```tsx
validate: {
  test1: xxx,
  test2: xxx,
  test3: xxx,
}
```

and set `criteriaMode` to `all`

**Reply by ElenaMazyrina** [reply]
If I understand correctly you suggest splitting my validation function into few. And in this case each of them will return one separate error message as string. 
But in my case I can't split my validation logic. My function is like a complex value parser: it tries to parse value and return bunch of messages if value in incorrect.
